src/index.js

```diff
@@ -37,18 +39,36 @@ const EmojiPicker = ({
   searchPlaceholder = null,
 }) => {
   const emojiListRef = useRef(null);
+  const emojiPickerAsideRef = useRef(null);
+  const emojiSearchRef = useRef(null);
+  const categoriesNavRef = useRef(null);
   const isMounted = useRef(true);
   const onClickRef = useRef(onEmojiClick);
 
   onClickRef.current = onEmojiClick;
 
+  const [handleKeyboard] = useKeyboardNavigation(
+    categoriesNavRef,
+    emojiSearchRef,
+    emojiListRef
+  );
+
   useEffect(
     () => () => {
       isMounted.current = false;
     },
     []
   );
 
+  useEffect(() => {
+    return tinykeys(window, {
+      ArrowUp: handleKeyboard,
+      ArrowDown: handleKeyboard,
+      ArrowLeft: handleKeyboard,
+      ArrowRight: handleKeyboard,
```

Comment:
Note that we're actually making these same checks in three places, and we're going to add more as we add more features

1. Here
2. in handleCategoriesKeyboard
3. in handSearchKeyboard

Now, I am thinking out loud. Wouldn't it be much nicer to have multiple effects, each subscribed to its own element, routing to the correct function? If we add these functions that I mentioned above, it should be a breeze:

```js
useEffect(() => {
  tinykeys(categoriesNavRef.current, {
    ArrowLeft: focusPrevCategory,
    ArrowRight: focusNextCategory,
    ArrowDown: focusNextSection,
  });
}, []);

useEffect(() => {
  tinykeys(emojiSearchRef.current, {
    ArrowRight: focusSkinTonePicker,
    ArrowDown: focusNextSection,
    ArrowUp: focusPrevSection,
  });
}, []);

// ...
```

This way we don't need to have a lot of code dedicated to routing, and just handle the logic declaratively. It's also wayyy safer, because we're only responding to events in the correct section, no need for further checks since tinykeys does them for us.